src/ng/q.js

```diff
@@ -636,6 +636,34 @@ function qFactory(nextTick, exceptionHandler, errorOnUnhandledRejections) {
     return deferred.promise;
   }
 
+  /**
+   * @ngdoc method
+   * @name $q#race
+   * @kind function
+   *
+   * @description
+   * Returns a promise that is resolved similarly as the first promise to resolve.
+   *
+   * @param {Array.<Promise>|Object.<Promise>} promises An array or hash of promises.
+   * @returns {Promise} Returns a promise that will be resolved as soon as the first promise in
+   *   `promises` is resolved. The resolved value will be the same as the value of the first
+   *   resolved promise.
+   */
+
+  function race(promises) {
+    var deferred = new Deferred();
+
+    forEach(promises, function(promise) {
+      when(promise).then(function(value) {
```

Comment:
Hmm, interesting. I am not too sure, but looking at the source for [`forEach`](https://github.com/angular/angular.js/blob/915e7a109c8bb512d6e31e639d898095be326c29/src/Angular.js#L237-L277), I don't think it will throw.